src/spawn.c

```diff
@@ -0,0 +1,1121 @@
+ *         in the command line exists and is executable.
+ *  @param error return location for error.
+ *
+ *  @return @c TRUE on success, @c FALSE if an error was set.
+ **/
+gboolean spawn_check_command(const gchar *command_line, gboolean execute, GError **error)
+{
+	gchar *program;
+
+#ifdef G_OS_WIN32
+	gboolean open_quote = FALSE;
+	const gchar *s, *arguments;
+
+	while (*command_line && strchr(BLANKS, *command_line))
+		command_line++;
+
+	if (!*command_line)
+	{
+		/* from glib */
+		g_set_error(error, G_SHELL_ERROR, G_SHELL_ERROR_EMPTY_STRING,
+			_("Text was empty (or contained only whitespace)"));
+		return FALSE;
+	}
+
+	if (*command_line == '"')
+	{
+		command_line++;
+		/* Windows allows "foo.exe, but we may have extra arguments */
+		if ((s = strchr(command_line, '"')) == NULL)
+		{
+			/* from glib, except for the fixed quote */
+			g_set_error(error, G_SHELL_ERROR, G_SHELL_ERROR_BAD_QUOTING,
+				_("Text ended before matching quote was found for %c."
+				  " (The text was '%s')"), '"', command_line);
+			return FALSE;
+		}
+
+		if (!strchr(BLANKS, s[1]))
+		{
+			g_set_error(error, G_SHELL_ERROR, G_SHELL_ERROR_BAD_QUOTING,
+				_("A quoted Windows program name must be entirely inside the quotes"));
+			return FALSE;
+		}
+	}
+	else
+	{
+		const gchar *quote = strchr(command_line, '"');
+
+		for (s = command_line; !strchr(BLANKS, *s); s++);
+
+		if (quote && quote < s)
+		{
+			g_set_error(error, G_SHELL_ERROR, G_SHELL_ERROR_BAD_QUOTING,
+				_("A quoted Windows program name must be entirely inside the quotes"));
+			return FALSE;
+		}
+	}
+
+	program = g_strndup(command_line, s - command_line);
+	arguments = s + (*s == '"');
+
+	for (s = arguments; *s; s++)
+	{
+		if (*s == '"')
+		{
+			const char *slash;
+
+			for (slash = s; slash > arguments && slash[-1] == '\\'; slash--);
+			if ((s - slash) % 2 == 0)
+				open_quote ^= TRUE;
+		}
+	}
+
+	if (open_quote)
+	{
+		/* from glib, except for the fixed quote */
+		g_set_error(error, G_SHELL_ERROR, G_SHELL_ERROR_BAD_QUOTING,
+			_("Text ended before matching quote was found for %c."
+			  " (The text was '%s')"), '"', command_line);
+		return FALSE;
+	}
+#else  /* G_OS_WIN32 */
+	int argc;
+	char **argv;
+
+	if (!g_shell_parse_argv(command_line, &argc, &argv, error))
+		return FALSE;
+
+	program = g_strdup(argv[0]);  /* note: empty string results in parse error */
+	g_strfreev(argv);
+#endif  /* G_OS_WIN32 */
+
+	if (execute && !g_find_program_in_path(program))
+	{
+		g_set_error(error, G_SHELL_ERROR, G_SHELL_ERROR_FAILED,  /* or SPAWN error? */
+			_("Program '%s' not found in path"), program);
+		g_free(program);
+		return FALSE;
+	}
+
+	g_free(program);
+	return TRUE;
+}
+
+
+/**
+ *  Kills a process.
+ *
+ *  @param pid id of the process to kill.
+ *  @param error return location for error.
+ *
+ *  @return @c TRUE on success, @c FALSE if an error was set.
+ **/
+gboolean spawn_kill_process(GPid pid, GError **error)
+{
+#ifdef G_OS_WIN32
+	if (!TerminateProcess(pid, 15))

```

Comment:
Oh, ok, my bad.  I assumed `TerminateProcess()` worked more or less like `kill()` and that the `15` was some kind of signal.  Now I checked the doc and I see it's the return value for the process, so OK anything non-0 (and better, non-1) is probably fine indeed.